### PR TITLE
Edge-Network-StaticRoutes-get/set/del/clean + UnitTests

### PR DESCRIPTION
**Pull request by Elchay-Levy**
**cterask.edge.netwok.py:**
Network.get_static_routes
Network.set_static_route
Network.del_static_route
Network.clean_all_static_routes

UnitTests
**cterasdk.tests.ut.test_edge_network.py:**
TestEdgeNetwork.test_set_static_routes
TestEdgeNetwork.test_set_static_routes_raise
TestEdgeNetwork.test_get_all_static_routes
TestEdgeNetwork.test_delete_static_route
TestEdgeNetwork.test_delete_static_route_raise
TestEdgeNetwork.test_clean_all_static_routes_success
TestEdgeNetwork.test_clean_all_static_routes_raise

### CODE REVIEW

File: cterasdk/edge/network.py [hunk 1]
```diff
@@ -176,3 +177,57 @@ def iperf(self, address, port=5201, threads=1, protocol=IPProtocol.TCP, directio
...
         except TaskError as error:
             return error.task.result.res
+
+    def get_static_routes(self):
+        """
+        Get all Static Routes
+        """
+        return self._gateway.get('/config/network/static_routes')
+
+    def set_static_route(self, source_ip, destination_ip, destination_mask):
```
Comment (saimonation) [inline cterasdk/edge/network.py:187]: Should be changed to "add_static_route" and "remove_static_route". "Set" suggests replacing the existing configuration alltogether. add/remove is consistent with other areas with similar functionality in the SDK.

File: cterasdk/edge/network.py [hunk 2]
```diff
@@ -176,3 +177,57 @@ def iperf(self, address, port=5201, threads=1, protocol=IPProtocol.TCP, directio
...
+            raise CTERAException('Static route deletion failed', error)
+
+    def clean_all_static_routes(self):
+        """
+        Clean all Static Routes
+        """
+        try:
+            self._gateway.execute('/config/network', 'cleanStaticRoutes')
+            logging.getLogger().info('All Static Routes cleaned Successfully')
+            return 'All Static Routes cleaned Successfully'
```
Comment (saimonation) [inline cterasdk/edge/network.py:230]: No need to return strings. If no errors occurred, then the command should return nothing

File: cterasdk/edge/network.py [hunk 3]
```diff
@@ -176,3 +177,57 @@ def iperf(self, address, port=5201, threads=1, protocol=IPProtocol.TCP, directio
...
+        :param str destination_ip: The destination IP
+        :param str destination_mask: The destination IP subnet mask CIDR Notation (1-32)
+        """
+        param = Object()
+        param.GwIP = source_ip
+        param.DestIpMask = f'{destination_ip}_{destination_mask}'
+        try:
+            res = self._gateway.add('/config/network/static_routes', param)
+            logging.getLogger().info(
+                "Static Route updated. %s", {'Source': param.GwIP, 'Destination': param.DestIpMask})
```
Comment (saimonation) [inline cterasdk/edge/network.py:201]: Lowercase r

File: cterasdk/edge/network.py [hunk 4]
```diff
@@ -176,3 +177,57 @@ def iperf(self, address, port=5201, threads=1, protocol=IPProtocol.TCP, directio
...
+        except CTERAException as error:
+            logging.getLogger().error("Static route creation failed.")
+            raise CTERAException('Static route creation failed', error)
+
+    def del_static_route(self, destination_ip, destination_subnet_mask):
+        """
+        Delete a Static Route
+
+        :param str destination_ip: The destination IP
+        :param str destination_subnet_mask: The destination IP subnet mask CIDR Notation (1-32)
```
Comment (saimonation) [inline cterasdk/edge/network.py:212]: Same comment as before. Collapse the two variables into one.
You should add a utils command that parses CIDR blocks and checks them if we don't have one already. The utils command can also validate it's a valid IP and that the CIDR block is a valid range, otherwise raise errors

Comment (Elchay-Levy) [reply cterasdk/edge/network.py]: Done, Thanks 
https://github.com/ctera/ctera-python-sdk/pull/156/commits/14cae0a36fc31c55dbeab38c0013ceb58a805d24

File: cterasdk/edge/network.py [hunk 5]
```diff
@@ -176,3 +177,57 @@ def iperf(self, address, port=5201, threads=1, protocol=IPProtocol.TCP, directio
...
+        """
+        return self._gateway.get('/config/network/static_routes')
+
+    def set_static_route(self, source_ip, destination_ip, destination_mask):
+        """
+        Set a Static Route
+
+        :param str source_ip: The source Filer IP (Gateway IP)
+        :param str destination_ip: The destination IP
+        :param str destination_mask: The destination IP subnet mask CIDR Notation (1-32)
```
[the remark this answers is not in the document]
Comment (Elchay-Levy) [reply cterasdk/edge/network.py]: Done, Thanks 
https://github.com/ctera/ctera-python-sdk/pull/156/commits/14cae0a36fc31c55dbeab38c0013ceb58a805d24

File: cterasdk/edge/network.py (hunk 3 above)
Comment (Elchay-Levy) [reply cterasdk/edge/network.py]: Done, Thanks 
https://github.com/ctera/ctera-python-sdk/pull/156/commits/14cae0a36fc31c55dbeab38c0013ceb58a805d24

File: cterasdk/edge/network.py [hunk 6]
```diff
@@ -176,3 +177,57 @@ def iperf(self, address, port=5201, threads=1, protocol=IPProtocol.TCP, directio
...
+            logging.getLogger().error("Static route deletion failed.")
+            raise CTERAException('Static route deletion failed', error)
+
+    def clean_all_static_routes(self):
+        """
+        Clean all Static Routes
+        """
+        try:
+            self._gateway.execute('/config/network', 'cleanStaticRoutes')
+            logging.getLogger().info('All Static Routes cleaned Successfully')
```
[the remark this answers is not in the document]
Comment (Elchay-Levy) [reply cterasdk/edge/network.py]: Done, Thanks 
https://github.com/ctera/ctera-python-sdk/pull/156/commits/14cae0a36fc31c55dbeab38c0013ceb58a805d24

File: tox.ini [hunk 7]
```diff
@@ -9,7 +9,7 @@ deps =
     -r{toxinidir}/docs/requirements.txt
     -r{toxinidir}/ut-requirements.txt
     flake8 >= 3.7.9
-    pylint >= 2.10.2
+    pylint >= 2.10.2,<=2.13.9
```
Comment (saimonation) [inline tox.ini:12]: Why did you need to reduce the pylint requirement? I recommend simply resolving and committing all errors you found in the newer version of pylint so we remain upstream as much as possible. I know it can be annoying but that's the approach we took so far.

File: cterasdk/edge/network.py [hunk 8]
```diff
@@ -176,3 +177,55 @@ def iperf(self, address, port=5201, threads=1, protocol=IPProtocol.TCP, directio
...
+            raise CTERAException('Static route creation failed', error)
+
+    def remove_static_route(self, destination_ip_mask):
+        """
+        Delete a Static Route
+
+        :param str destination_ip_mask: The destination IP and CIDR block (10.0.0.0/16)
+        """
+        try:
+            ip_mask = destination_ip_mask.replace("/", "_")
```
Comment (saimonation) [inline cterasdk/edge/network.py:213]: Python3 includes an ipaddress module. Consider using this:
https://stackoverflow.com/questions/45988215/python-how-to-validate-an-ip-address-with-a-cidr-notation-import-socket
This will also raise errors

[the remark this answers is not in the document]
Comment (saimonation) [reply cterasdk/edge/network.py:213]: ``import ipaddress``
``ipaddr = ipaddress.ip_network('192.168.0.1/32')  # catch ValueError log and raise an error``
``string = f'{ipaddr.network_address}_{ipaddr.prefixlen}'``

[the remark this answers is not in the document]
Comment (saimonation) [reply cterasdk/edge/network.py:213]: It's probably beneficial to include this in the common.utils module so other modules can use it.. we use IP's quite extensively.

File: tox.ini (hunk 7 above)
Comment (Elchay-Levy) [reply tox.ini]: Pylint deprecated options were removed and resolved, and I removed the max version limit.
https://github.com/ctera/ctera-python-sdk/pull/156/commits/1091372d41deb0ae6fbc58d96b09d95aa73a96bd

File: cterasdk/edge/network.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (Elchay-Levy) [reply cterasdk/edge/network.py]: Done, created new Utility parse_to_ipaddress, and validated address type, added to add_static_route and remove_static_route.
https://github.com/ctera/ctera-python-sdk/commit/1091372d41deb0ae6fbc58d96b09d95aa73a96bd
